lumicks/pylake/population/mixture.py

```diff
@@ -0,0 +1,178 @@
+        """
+        return cls(slc, K, init_method="kmeans", n_init=1, tol=1e-3, max_iter=100)
+
+    @property
+    def exit_flag(self):
+        """Model optimization information."""
+        return {
+            "converged": self._model.converged_,
+            "n_iter": self._model.n_iter_,
+            "lower_bound": self._model.lower_bound_,
+        }
+
+    @property
+    def _map(self):
+        """Indices of sorted means."""
+        return np.argsort(self._model.means_.squeeze())
+
+    @property
+    @as_sorted
+    def weights(self):
+        """Model state weights."""
+        return self._model.weights_
+
+    @property
+    @as_sorted
+    def means(self):
+        """Model state means."""
+        return self._model.means_.squeeze()
+
+    @property
+    @as_sorted
+    def variances(self):
+        """Model state variances."""
+        return self._model.covariances_.squeeze()
+
+    @property
+    def std(self):
+        """Model state standard deviations."""
+        return np.sqrt(self.variances)
+
+    def label(self, trace):
+        """Label channel trace data as states."""
+        data = trace.data.reshape((-1, 1))
+        labels = self._model.predict(data)  # wrapped model labels
+        output_states = np.argsort(self._map)  # output model state labels in wrapped model order
+        return output_states[labels]  # output model labels
+
+    def bic(self):
+        """Bayesian Information Criterion."""
+        return self._model.bic(self._training_data.data.reshape((-1, 1)))
+
+    def aic(self):
+        """Akaike Information Criterion."""
+        return self._model.aic(self._training_data.data.reshape((-1, 1)))
+
+    def pdf(self, x):
+        """Probability Distribution Function (states as rows)."""
+        components = np.vstack([stats.norm(m, s).pdf(x) for m, s in zip(self.means, self.std)])
+        return self.weights.reshape((-1, 1)) * components
+
+    def hist(self, trace=None, n_bins=100, plot_kwargs={}, hist_kwargs={}):
+        """Plot a histogram of the data overlaid with the model PDF.
+
+        Parameters
+        ----------
+        trace : Slice-like
+            data object to histogram; default is training data.
+        n_bins : int
+            number of histogram bins
+        plot_kwargs : dict
+            plotting keyword arguments passed to the PDF line plot
+        hist_kwargs : dict
+            plotting keyword arguments passed to the histogram plot
+        """
+        import matplotlib.pyplot as plt
+
+        if trace is None:
+            trace = self._training_data
+
+        hist_kwargs = {"facecolor": "#c5c5c5", **hist_kwargs}
+
+        lims = (np.min(trace.data), np.max(trace.data))
+        bins = np.linspace(*lims, num=n_bins)
+        x = np.linspace(*lims, num=(n_bins * 5))
+
+        g_components = self.pdf(x)
+
+        plt.hist(trace.data, bins=bins, density=True, **hist_kwargs)
+        # reset color cycle
+        plt.gca().set_prop_cycle(None)
```

Comment:
Why is this needed?

Reply:
I noticed this when testing and I was plotting both the histogram and labeled data in the same figure. The first call to `plt.hist()` causes the color cycle to increment, however this doesn't happen inside the `GMM.plot()` method (I think because `trace.plot()` is used instead of plotting the data manually). 

At any rate, this ensures that the same state has the same color if using the default color cycle in plots generated either from `GMM.hist()` or `GMM.plot()`. I don't think it's absolutely necessary, but I thought it was a nice feature and only requires 1 line of code.